jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/Extension.java

```diff
@@ -23,6 +23,13 @@ public interface Extension extends IncomingFrames, OutgoingFrames
 
     void init(ExtensionConfig config, WebSocketComponents components);
 
+    /**
+     * Used to clean up any resources after connection close.
+     */
+    default void close()
```

Comment:
Should `Extension extend Closeable`? 
So that `ExtensionStack` can call `IO.close(extension)` and not worry about exceptions thrown by `close()`.